cmd/erigon-cl/core/state/raw/copy.go

```diff
@@ -0,0 +1,72 @@
+package raw
+
+import (
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon/cl/clparams"
+	"github.com/ledgerwatch/erigon/cl/cltypes"
+)
+
+func (b *BeaconState) CopyInto(dst *BeaconState) error {
+	dst.genesisTime = b.genesisTime
+	dst.genesisValidatorsRoot = b.genesisValidatorsRoot
+	dst.slot = b.slot
+	dst.fork = b.fork.Copy()
+	dst.latestBlockHeader = b.latestBlockHeader.Copy()
+	copy(dst.blockRoots[:], b.blockRoots[:])
+	copy(dst.stateRoots[:], b.stateRoots[:])
+	dst.historicalRoots = make([]libcommon.Hash, len(b.historicalRoots))
+	copy(dst.historicalRoots, b.historicalRoots)
+	dst.eth1Data = b.eth1Data.Copy()
+	dst.eth1DataVotes = make([]*cltypes.Eth1Data, len(b.eth1DataVotes))
+	for i := range b.eth1DataVotes {
+		dst.eth1DataVotes[i] = b.eth1DataVotes[i].Copy()
+	}
+	dst.eth1DepositIndex = b.eth1DepositIndex
+	dst.validators = make([]*cltypes.Validator, len(b.validators))
+	for i := range b.validators {
+		dst.validators[i] = b.validators[i].Copy()
```

Comment:
does it not kill the purpose of CopyInto, you use the pre-existent validators and just assign it. after CopyInto can take advantage of pre-existic alloc by beacon state instead of making new ones